### PR TITLE
Upgrade openvpn plugin

### PR DESCRIPTION
**Pull request by faern**
We are finally going to release `openvpn-plugin 0.4.0` and depend on that. We have had lots of improvements in the git repo of that crate for a long time and it never reached crates.io. We want others to be able to use it and we want to stop depending on git as well.

Git checklist:

* [ ] Describe the change in **`CHANGELOG.md`** under the `[Unreleased]` header.
* [x] Check that commits follow the [Mullvad coding guidelines](https://github.com/mullvad/coding-guidelines)

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2192)

### CODE REVIEW

File: talpid-core/Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ rand = "0.7"
 
 
 [target.'cfg(not(target_os="android"))'.dependencies]
-openvpn-plugin = { git = "https://github.com/mullvad/openvpn-plugin-rs", branch = "auth-failed-event", features = ["serde"] }
+openvpn-plugin = { git = "https://github.com/mullvad/openvpn-plugin-rs", rev = "d33b0d2795b99c83804d4c757acef881b901a5d0", features = ["serde", "auth-failed-event"] }
```
Comment (faern) [inline talpid-core/Cargo.toml:34]: This is temporary. I will publish `0.4.0` and depend on that once this PR has been approved. I wanted this code change to get approved before I went ahead and actually published anything.

File: talpid-core/src/tunnel/openvpn.rs [hunk 2]
```diff
@@ -684,12 +685,13 @@ mod event_server {
             &self,
             request: Request<EventType>,
         ) -> std::result::Result<Response<()>, tonic::Status> {
-            log::info!("OpenVPN event {:?}", request);
-
             let request = request.into_inner();
+            log::debug!("OpenVPN event {:?}", request);
```
Comment (faern) [inline talpid-core/src/tunnel/openvpn.rs:689]: We should not have debug representations (`{:?}`) in an `info` level log. Debug representations should only exist in `debug` level and more detailed. I also felt we could do this logging *after* calling `into_inner` in order to not log a bunch of gRPC headers we don't care about.

File: talpid-core/src/tunnel/openvpn.rs [hunk 3]
```diff
@@ -684,12 +685,13 @@ mod event_server {
...
 
-            let event_type = openvpn_plugin::EventType::try_from(request.event)
-                .ok_or(tonic::Status::invalid_argument("Unknown event type"))?;
+            let event_type =
+                openvpn_plugin::EventType::try_from(request.event).map_err(|event: i32| {
+                    tonic::Status::invalid_argument(format!("Unknown event type: {}", event))
```
Comment (faern) [inline talpid-core/src/tunnel/openvpn.rs:693]: Not sure if this has any practical difference. But with the new way `try_from` works we can get the invalid integer back, so we can add it to the error message.

**Review by dlon — APPROVED** [review]
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2192)* status: all files reviewed, 3 unresolved discussions (waiting on @faern)

___
*[talpid-core/src/tunnel/openvpn.rs, line 689 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2192#-MK4enC4G7IOJZDLu9Ip-r1-689:-MK4hTA4D8mGo9tXS2kY:bdcv6l8) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/85191c20cb563d294cf2f2974cbc10bb5fa43d92/talpid-core/src/tunnel/openvpn.rs#L689)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

We should not have debug representations (`{:?}`) in an `info` level log. Debug representations should only exist in `debug` level and more detailed. I also felt we could do this logging *after* calling `into_inner` in order to not log a bunch of gRPC headers we don't care about.
</blockquote></details>

Good point.

**Comment by faern** [comment]
This PR was superseded by #2278

**Pull request closed without merging**